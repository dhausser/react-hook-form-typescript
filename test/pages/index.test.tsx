import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import Index from '../../pages/index'

describe('Home page', () => {
  it('sign up page title', () => {
    const { getByText } = render(<Index />)
    const titleElement = getByText('Sign Up')
    console.log(titleElement)
    expect(titleElement).toBeInTheDocument()
  })

  it('first name input field', () => {
    const { getByText } = render(<Index />)
    const inputElement = getByText('First Name')
    console.log(inputElement)
    expect(inputElement).toBeInTheDocument()
  })

  it('last name input field', () => {
    const { getByText } = render(<Index />)
    const inputElement = getByText('Last Name')
    console.log(inputElement)
    expect(inputElement).toBeInTheDocument()
  })

  it('gender input field', () => {
    const { getByText } = render(<Index />)
    const inputElement = getByText('Gender')
    console.log(inputElement)
    expect(inputElement).toBeInTheDocument()
  })

  it('username input field', () => {
    const { getByText } = render(<Index />)
    const inputElement = getByText('Username')
    console.log(inputElement)
    expect(inputElement).toBeInTheDocument()
  })

  it('username input field', () => {
    const { getByText } = render(<Index />)
    const inputElement = getByText('Username')
    console.log(inputElement)
    expect(inputElement).toBeInTheDocument()
  })

  it('email input field', () => {
    const { getByText } = render(<Index />)
    const inputElement = getByText('Email')
    console.log(inputElement)
    expect(inputElement).toBeInTheDocument()
  })

  it('age input field', () => {
    const { getByText } = render(<Index />)
    const inputElement = getByText('Age')
    console.log(inputElement)
    expect(inputElement).toBeInTheDocument()
  })

  it('about you input field', () => {
    const { getByText } = render(<Index />)
    const inputElement = getByText('About you')
    console.log(inputElement)
    expect(inputElement).toBeInTheDocument()
  })

  it('matches snapshot', () => {
    const { asFragment } = render(<Index />, {})
    expect(asFragment()).toMatchSnapshot()
  })

  it('clicking button triggers alert', () => {
    const { getByText } = render(<Index />, {})
    window.alert = jest.fn()
    fireEvent.click(getByText('Test Button'))
    expect(window.alert).toHaveBeenCalledWith('With typescript and Jest')
  })
})
