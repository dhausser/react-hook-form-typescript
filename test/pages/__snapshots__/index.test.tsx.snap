// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="App"
  >
    <h1>
      Sign Up
    </h1>
    <button>
      Test Button
    </button>
    <label>
      First Name
    </label>
    <input
      name="firstName"
    />
    <label>
      Last Name
    </label>
    <input
      name="lastName"
    />
    <label>
      Gender
    </label>
    <select
      name="gender"
    >
      <option>
        Select...
      </option>
      <option>
        Male
      </option>
      <option>
        Female
      </option>
    </select>
    <label>
      Username
    </label>
    <input
      name="username"
    />
    <label>
      Email
    </label>
    <input
      name="email"
    />
    <label>
      Age
    </label>
    <input
      name="age"
      type="number"
    />
    <label>
      About you
    </label>
    <textarea
      name="about"
    />
    <input
      type="submit"
    />
  </form>
</DocumentFragment>
`;
